File: frontend/lib/models/satellite.dart
// lib/models/satellite.dart — Satellite data model.
//
// Fields:
//   double soilMoisture, double temperature, double rainProbability,
//   double ndvi, double windSpeed, String timestamp
//
// Responsibilities:
//   - Define a SatelliteData class with a fromJson() factory constructor.
//   - JSON keys match the Go backend GET /api/satellite response.
//
// Used by:
//   services/satellite_service.dart — deserializes API response.
//   screens/dashboard/ — displays current environmental data.
//   widgets/satellite_card.dart — renders the satellite data card.
class SatelliteData {}

File: backend/internal/models/satellite.go
package models

type SatelliteData struct {
	ID              int     `json:"id"`
	FarmID          int     `json:"farm_id"`
	SoilMoisture    float64 `json:"soil_moisture"`
	Temperature     float64 `json:"temperature"`
	RainProbability float64 `json:"rain_probability"`
	NDVI            float64 `json:"ndvi"`
	WindSpeed       float64 `json:"wind_speed"`
	Source          string  `json:"source"`
	Timestamp       string  `json:"timestamp"`
}
